random_generator/utils/transitions.py

```diff
@@ -0,0 +1,35 @@
+import queue as python_queue
+import random
+
+from utils.configuration import Config
+from utils.utility_functions import get_move_vector, get_x_final_transitions, mul
+
+
+def transition_generator(config: Config):
+    visited = set()
+    depth = config.depth_size
+    queue = python_queue.Queue()
+    queue.put(0)
+    visited.add(0)
+    while not queue.empty():
+        transitions_to: dict = {}
+        q = queue.get()
+        this_state_moves = [
+            random.randint(1, config.max_num_moves)
+            for _ in range(config.number_of_players)
+        ]
+        final_transitions = get_x_final_transitions(random, depth, mul(this_state_moves))
```

Comment:
So, by looking at the definition of `get_x_final_transitions`, it seems like `depth` as actually the size of the state space, and that the return value of the function is a destination states for the transitions. In that case I think the naming should be changed.

Reply:
Yes, good point concerning the naming of the method.

Though regarding if I could find the destination one at the time, in the implementation that I've created it requires a list of states that each move results in. Because this is the way i "design" the move vectors.

It might be a way to improve performance, later, to just have it yield the destination, as we could wait to _just_ before writing to the json file, to actually generate the number. But the performance boost would be so small if at all, that I don't consider it worth spending time on. :)

Maybe you should create an issue, and we can ice-box it?

Reply:
Let me clarify what I meant with "You could also just find the destination state one at the time". I simply just noticed that `get_x_final_transitions` creates a list using list comprehension and then you iterate through that list immediate after. This means it could be reduced to:
```python
for _ in range(mul(this_state_moves)):
    # inlined get_x_final_transitions/get_resulting_states
    x = random.randint(0, max_trans - 1)
    if transitions_to.get(q) is None:
        transitions_to[q] = []
    transitions_to[q].append(x)
    if x not in visited:
        visited.add(x)
        queue.put(x)
```
It does exactly the same. Your way of doing it is not a problem at all (unless `mul(this_state_moves)` is a HUGE number).

Reply:
Ah, yes, I get what you mean.
I have corrected this :)

Reply:
I think you actually broke it. The semantics are different now 😅

Reply:
Yep, that was a brain fart :) #FixD